[15.0][FIX+IMP] l10n_es_aeat_mod592: Compare if no weight with the correct decimal precision + Misc changes

victoralmau opened the pull request:
Cambios hechos:
- [x] Comprar si el peso no es correcto con la precisión decimal correspondiente.
- [x] Definir digits a los campos `kgs`, `no_recycling_kgs`, `product_uom_qty`
- [x] Definir datos correctos en el excel que se exporta
- [x] Posibilidad de exportar csv

Por favor @pedrobaeza puedes revisarlo?

@Tecnativa TT52290 TT52986

victoralmau committed:
[FIX] l10n_es_aeat_mod592: Compare if no weight with the correct decimal precision

TT52290

pedrobaeza commented:
En el BOE exportado, ¿cuántos decimales se tienen? ¿Es la misma precisión?

victoralmau commented:
> En el BOE exportado, ¿cuántos decimales se tienen? ¿Es la misma precisión?

Actualmente no hay configuración para exportar a BOE. 
Quizás también sería recomendable definir en los campos correspondientes la precisión decimal de peso y mostrar el dato adecuado en pantalla https://github.com/OCA/l10n-spain/blob/4da2f26d578bcd4dada5475f4839883193b04f01/l10n_es_aeat_mod592/models/mod592_line_mixin.py#L44

pedrobaeza commented:
Bueno, aunque no haya exportación, es importante saber cómo lo exportaría, porque ésa es la precisión que va a marcar el mínimo exportable, y la que habría que utilizar para todo.

victoralmau committed:
[IMP] l10n_es_aeat_mod592: Definir digits a los campos kgs, no_recycling_kgs, product_uom_qty

TT52290

victoralmau commented:
Veo que hay una opción de export_type=float en las líneas de exportación, aunque en todos se define `decimal_size` por lo que creo no es muy importante en este caso el como se defina el campo. https://github.com/OCA/l10n-spain/blob/8ff75677bf227a05b1f7ff90c7d526aee32612ce/l10n_es_aeat/wizard/export_to_boe.py#L242 En cualquier caso he añadido el digits correspondiente también a los campos para que los datos que se muestren por pantalla sean coherentes.

victoralmau committed:
[IMP] l10n_es_aeat_mod592: Definir datos correctos en el excel que se exporta

Cambios hechos:
- Se añade el concepto y el product_description al método _get_csv_report_info() del mixin (necesario en ambos)
- Se añade el campo product_description al mixin (necesario en ambos)
- Los valores de Kilogramos y Kilogramos No Reciclados se deben definir con , como separador decimal
- Definir el valor del campo Fecha Hecho Contabilizado en formato: DD-MM-YYYY
- Adquiriente: Se define como Descripción Producto la descripción del producto en lugar de fiscal_acquirer
- Adquiriente: Se añade la columna Régimen fiscal (fiscal_acquirer)  después de la descripción de producto
- Adquiriente: Se mueven los campo Prov./Dest. al lugar adecuado, antes de Kilogramos

TT52986

pedrobaeza commented:
Entiendo que falta como mínimo la exportación a CSV, ¿verdad?

victoralmau committed:
[IMP] l10n_es_aeat_mod592: Posibilidad de exportar csv

TT52986

victoralmau commented:
En un principio ya estaría listo para fusionarse (no hay mas cambios pendientes).

victoralmau commented:
Ping @pedrobaeza

pedrobaeza reviewed (approved):
/ocabot merge major
